### PR TITLE
Update debian_swraid_lvm_luks.md

### PR DESCRIPTION
**Pull request by wnhre2ur8cxx8**
Refactored, simplified and properly tested.

#28

### CODE REVIEW

File: debian/debian_swraid_lvm_luks.md [hunk 1]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
 
 - Boot to the rescue system via hetzners server management page
 - install a minimal Debian 9.4 with hetzners "installimage" skript (https://wiki.hetzner.de/index.php/Installimage)
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:18]: I've just run this guide on Debian 10.7 - so we can safely add 10.7 here

File: debian/debian_swraid_lvm_luks.md [hunk 2]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
...
-- I choosed the following logical volumes on my system to keep it simple:
+- I chose the default raid1/lvm setup to keep it simple:
 
 ```
-lv-home (60GB) ext4
-lv-log (30GB) ext4
-lv-swap (10GB) swap
-lv-root (all) -> means remaining space ext4
+cat configs/simple-debian64-raid-lvm
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:22]: Is this hetzner specific? If yes, we should probably keep the old configuration block - however, a hint to the default setup seems reasonable to me

File: debian/debian_swraid_lvm_luks.md [hunk 3]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
...
-LV vg0 swap swap swap 4G
-LV vg0 log /log ext4 30g
-LV vg0 root / ext4 all
-```
-And change the partition section accordingly
-```
-PART /boot ext3 512M
-PART lvm vg0 all
-```
-- after you adjusted all parameters in the install config file, press F10 to install the debian minimal system
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:39]: See previous comment

File: debian/debian_swraid_lvm_luks.md [hunk 4]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
...
-- install busybox and dropbear
-- `apt update && apt install busybox dropbear lvm2`
-- Edit your `/etc/initramfs-tools/initramfs.conf` and set `BUSYBOX=y`
-- Create a new ssh key for unlocking your encrypted volumes when it is rebooting
-- `ssh-keygen -t rsa -b 4096 -f .ssh/dropbear`
-- Create the needed folders for dropbear keys
-- run `vi /etc/dropbear-initramfs/authorized_keys`
-- Paste your pub key `.ssh/dropbear.pub` in there
+- connect via ssh-key you chose before for the rescue image (attention to the .ssh/known_hosts file..)
+- edit `/etc/ssh/sshd_config` and set `#PasswordAuthentication yes` to `PasswordAuthentication no
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:31]: Isn't this a bit confusing here to setup ssh authentication for (as far as I can tell) the _real_ server?

File: debian/debian_swraid_lvm_luks.md [hunk 5]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
...
+- change your root password: `passwd`
+- update and install dropbear-initramfs
+  - `apt update && apt upgrade && apt install dropbear-initramfs`
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:34]: Was not aware of this meta-package - nice one :+1:

File: debian/debian_swraid_lvm_luks.md [hunk 6]
```diff
@@ -16,41 +16,45 @@ This guide *could* work at any other provider with a rescue system.
...
+   IdentityFile ~/.ssh/id_dropbear
+   User root
+   UserKnownHostsFile ~/.ssh/known_hosts.initramfs
+ ```
+ 
+- Back on the server create the needed folders for dropbear keys
+- Create and edit `/etc/dropbear-initramfs/authorized_keys`:
+
+```
+no-port-forwarding,no-agent-forwarding,no-X11-forwarding,command="/bin/cryptroot-unlock" <paste the public key of your local computer here (.ssh/id_dropbear)>
```
Comment (sirtoobii) [inline debian/debian_swraid_lvm_luks.md:52]: Maybe a hint where the user is supposed to find the _local_ public-key is useful here

**Review by sirtoobii — COMMENTED** [review]
Very nice idea to simplify this tutorial. 
As mentioned above, I would keep the "complicated"  volume configurations  since I would not expect that a) all users of this guide are hetzner customers and b) there are maybe some "intermediate" users which have their own requirements regarding the volume setup.

File: debian/debian_swraid_lvm_luks.md (hunk 2 above)
Comment (wnhre2ur8cxx8) [reply debian/debian_swraid_lvm_luks.md:22]: Yes it is!

File: debian/debian_swraid_lvm_luks.md (hunk 3 above)
Comment (wnhre2ur8cxx8) [reply debian/debian_swraid_lvm_luks.md:39]: again :)

**Comment by wnhre2ur8cxx8** [comment]
> Very nice idea to simplify this tutorial.
> As mentioned above, I would keep the "complicated" volume configurations since I would not expect that a) all users of this guide are hetzner customers and b) there are maybe some "intermediate" users which have their own requirements regarding the volume setup.

My take on it was that b) intermediate user any have the ability to adjust it if they want, so stick to uncomplicated and safe defaults and a) this repo is named "disk-encrpytion-hetzner" so I expected the exact opposite :)

File: debian/debian_swraid_lvm_luks.md (hunk 5 above)
Comment (wnhre2ur8cxx8) [reply debian/debian_swraid_lvm_luks.md:34]: it works super well, gotta love debian :)

File: debian/debian_swraid_lvm_luks.md (hunk 4 above)
Comment (wnhre2ur8cxx8) [reply debian/debian_swraid_lvm_luks.md:31]: True! We should let the users decide if they want to. Maybe add an "if you want to disable password authentication for better security, you can edit ..."

File: debian/debian_swraid_lvm_luks.md (hunk 6 above)
Comment (wnhre2ur8cxx8) [reply debian/debian_swraid_lvm_luks.md:52]: so something like `(most likely in your home folder under .ssh/id_dropbear.pub)`

Also I messed up big here and forgot the .pub. Maybe add a warning to never paste the private key anywhere?

**Comment by Biepa** [comment]
Hey, I really like the simplified writeup. Thanks for that.
I have a question. If I want to use just one disk for the system instead of two, what would i do on the line, where it says "mount /dev/md0 /mnt/boot"?
I could follow the tutorial up to this point.

**Comment by wnhre2ur8cxx8** [comment]
> Hey, I really like the simplified writeup. Thanks for that. I have a question. If I want to use just one disk for the system instead of two, what would i do on the line, where it says "mount /dev/md0 /mnt/boot"? I could follow the tutorial up to this point.

You mean If you dont use a raid setup? Or you mean without a seperate /boot partition? You can try to setup it up with an encrypted /boot partition, but its harder. And you would still need a seperate partiton. See this wiki article for some explanations: https://wiki.archlinux.org/title/Dm-crypt/Encrypting_an_entire_system#LUKS_on_LVM

**Comment by Biepa** [comment]
@wnhre2ur8cxx8 Yes, i mean, that I have two disks, but dont want a raid setup.
I currently get up to the filesystem creation, but "/dev/vg0/root" doesnt exist there anymore.

**Comment by Biepa** [comment]
The only thing that I changed from your writeup until there is:
`cryptsetup luksFormat /dev/md1`
i changed this to
`cryptsetup luksFormat /dev/nvme0n1p2`

while having this as the state after the install without raid
grafik

**Comment by Biepa** [comment]
I also tried doing an encypted installation directly via hetzner with

```
CRYPTPASSWORD <pw here>
DRIVE1 /dev/nvme0n1
BOOTLOADER grub
HOSTNAME name123
PART /boot ext4 5G
PART /     ext4 all crypt
IMAGE /root/images/Debian-1100-bullseye-amd64-base.tar.gz
```

and then more or less following the instructions here. Everything seems fine until reboot. I even can connect to dropbear but I get "sh: cryptroot-unlock: not found" and then the session is closed.

**Comment by Biepa** [comment]
Just got it.

File: debian/debian_swraid_lvm_luks.md (hunk 5 above)
Comment (oxivanisher) [reply debian/debian_swraid_lvm_luks.md:34]: Attention, I also had to install `cryptsetup-initramfs` or `cryptroot-unlock` was missing.

**Comment by oxivanisher** [comment]
My PV size was wrong after I followed this version of the guide.
```
  WARNING: Device /dev/mapper/cryptroot has size of 7812687488 sectors which is smaller than corresponding PV size of 7812720256 sectors. Was device resized?
  WARNING: One or more devices used as PVs in VG vg0 have changed sizes.
```
It was a easy fix (probably since I have some free space in my VG) with:
`pvresize /dev/mapper/cryptroot`

**Comment by nc7s** [comment]
I've successfully followed this guide to encrypt my server, thanks for the work. It seems, however, that installing `dropbear-initramfs` is sort of a must, because installing `busybox` and `dropbear` alone didn't make it, despite my strictly following the guide. And installing that makes it work as promised.

**Pull request closed without merging**